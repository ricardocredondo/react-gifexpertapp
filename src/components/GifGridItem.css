.gifgriditem {
  margin: 0.8em;
  text-align: center;
  border: 2px solid #fff;
  border-radius: 5px;
  color: aliceblue;
  /* flex: 2 2 auto; */
}
.gifgriditem__img {
  width: 100%;
  vertical-align: top;
}
.gifgriditem__text {
  padding: 0.4em 0;
}
