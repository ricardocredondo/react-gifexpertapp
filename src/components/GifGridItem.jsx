import React from 'react';
import PropTypes from 'prop-types';
import './GifGridItem.css';

const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="gifgriditem animate__animated animate__backInDown">
      <img src={url} className="gifgriditem__img" alt={title} key={id} />
      <p className="gifgriditem__text">{title}</p>
    </div>
  );
};
GifGridItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
export default GifGridItem;
