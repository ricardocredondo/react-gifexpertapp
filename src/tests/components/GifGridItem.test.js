import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';
describe('Pruebas en <GifGridItem />', () => {
  const [id, title, url] = [
    'alskdjf443j',
    'la perra vieja',
    'http://localhost',
  ];
  const wrapper = shallow(<GifGridItem id={id} title={title} url={url} />);
  test('debe mostrar <GifGridItem /> correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('el elemento img debe tener la clase gifgriditem__img', () => {
    const clase = 'gifgriditem__img';
    const claseElemento = wrapper.find('img').hasClass(clase);
    expect(claseElemento).toBe(true);
  });
  test('la imagen debe tener atributos src y alt igual url y title respectivamente', () => {
    const img = wrapper.find('img');
    const src = img.prop('src');
    const alt = img.prop('alt');
    const key = img.key();
    expect(src).toBe(url);
    expect(alt).toBe(title);
    expect(key).toBe(id);
  });

  test('debe mostrar un parrafo con title', () => {
    const parrafo = wrapper.find('p').text();
    expect(parrafo).toBe(title);
  });
});
