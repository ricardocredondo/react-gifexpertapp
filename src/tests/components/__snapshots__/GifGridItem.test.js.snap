// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="gifgriditem animate__animated animate__backInDown"
>
  <img
    alt="la perra vieja"
    className="gifgriditem__img"
    key="alskdjf443j"
    src="http://localhost"
  />
  <p
    className="gifgriditem__text"
  >
    la perra vieja
  </p>
</div>
`;
